### PR TITLE
Fall back to alternate way of detecting minor version of Nvidia drivers if needed

### LINKED ISSUE #4617: GPU passthrough fails if Nvidia device minor is missing
OS: Ubuntu 18.04
[lxc-info.txt](https://github.com/lxc/lxd/files/2064229/lxc-info.txt)

Users of older graphics cards have to use an older driver... which doesn't provide the minor number where lxd expects it.
#4441 added a nicer error message, but users with old drivers are still left scratching their heads.

### PR DESCRIPTION
**Pull request by dankegel**
findNvidiaMinorOld() only handles the most common case; it's a bit of a placeholder.

Fixes #4617 for case of a single nvidia card (probably enough to reduce user complaints)

Very lightly tested.  This is my first go.

### CODE REVIEW

File: lxd/devices.go [hunk 1]
```diff
@@ -101,6 +101,65 @@ type cardIds struct {
...
+func findNvidiaMinorOld() (string, error) {
+	var minor string;
+
+	// Handle single nvidia card with driver too old to report device minor
+	ents, err := ioutil.ReadDir("/dev")
+	if err != nil {
+		return "", err
+	}
+	for _, ent := range ents {
+		rp := regexp.MustCompile("^nvidia([0-9]+)$")
```
Comment (stgraber) [inline lxd/devices.go]: Why compile this at every iteration of the loop?

File: lxd/devices.go [hunk 2]
```diff
@@ -101,6 +101,65 @@ type cardIds struct {
 	pci string
 }
 
+// Fallback for old drivers which don't provide "Device Minor:"
+func findNvidiaMinorOld() (string, error) {
+	var minor string;
```
Comment (stgraber) [inline lxd/devices.go]: No need for semicolons in Go

File: lxd/devices.go [hunk 3]
```diff
@@ -101,6 +101,65 @@ type cardIds struct {
 	pci string
 }
 
+// Fallback for old drivers which don't provide "Device Minor:"
+func findNvidiaMinorOld() (string, error) {
```
Comment (brauner) [inline lxd/devices.go]: Does this presuppose that there is not just one Nvidia card but also one `/dev/dri/card<idx>` card?

Comment (dankegel) [reply lxd/devices.go]: I don't think so, but I wouldn't know.  I did test a bit on two machines with two entries in /dev/dri/card* (an nvidia card and builtin intel graphics) and it seemed to work fine with the old and new nvidia cards I tried.

**Comment by stgraber** [comment]
One minor nit, Nvidia is actually capitalized NVIDIA. I'll update the branch, send through Jenkins and if all good, will merge.

**Comment by stgraber** [comment]
jenkins: test this please

**Pull request merged**